pinot-common/src/main/java/org/apache/pinot/common/utils/DataSchema.java

```diff
@@ -345,6 +345,9 @@ public DataType toDataType() {
      * compatible with the type.
      */
     public Serializable convert(Object value) {
+      if (value == null) {
```

Comment:
Consider doing the null check on the caller side. The passed in value should always be non-null